widget/entry_cursor_anim.go

```diff
@@ -0,0 +1,156 @@
+package widget
+
+import (
+	"image/color"
+	"sync"
+	"time"
+
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/canvas"
+	"fyne.io/fyne/v2/theme"
+)
+
+type safeCounter struct {
+	mu  sync.RWMutex
+	cnt int
```

Comment:
Doesn't `cnt` sound bad?

Reply:
Yeah, I will rename it to `val`, is it better?